### PR TITLE
Bug 1914127: Delete the ingress.openshift.io/operator finalizer

### PR DESCRIPTION
**Pull request by Miciah**
Add logic to remove the operator's finalizer on LoadBalancer services, which is no longer needed to clean up DNS records since the DNSRecord CRD was added.

* `pkg/operator/controller/ingress/load_balancer_service.go` (`ensureLoadBalancerService`): Use the new `deleteLoadBalancerService` method to delete the service as needed.  Use the new `deleteLoadBalancerServiceFinalizer` method to delete any finalizer on any existing service in the delete and update cases.
(`desiredLoadBalancerService`): Delete logic to add a finalizer to the service.
(`deleteLoadBalancerServiceFinalizer`): New method.  Delete any finalizer from the service.
(`finalizeLoadBalancerService`): Refactor to use the new `deleteLoadBalancerServiceFinalizer` method.
(`deleteLoadBalancerService`): New method.

### CODE REVIEW

File: pkg/operator/controller/ingress/load_balancer_service.go [hunk 1]
```diff
@@ -292,6 +324,18 @@ func (r *reconciler) createLoadBalancerService(service *corev1.Service) error {
 	return nil
 }
 
+// deleteLoadBalancerService deletes a load balancer service.
+func (r *reconciler) deleteLoadBalancerService(service *corev1.Service, options *crclient.DeleteOptions) error {
+	if err := r.client.Delete(context.TODO(), service, options); err != nil {
```
Comment (sgreene570) [inline pkg/operator/controller/ingress/load_balancer_service.go:329]: Is there a foreseeable reason to include an `*crclient.DeleteOptions` parameter? I'm not opposed to adding one here, but I recognize that the other `r.client.Delete` calls in this repo exclude this paramter.

**Review by sgreene570 — COMMENTED** [review]
Overall LGTM, I just have one small question.

File: pkg/operator/controller/ingress/load_balancer_service.go (hunk 1 above)
Comment (Miciah) [reply pkg/operator/controller/ingress/load_balancer_service.go:329]: Good question, and I had to go back to see why I did this.  Yes, we will want to specify `PropagationPolicy: "Foreground"` when deleting and recreating the service if/when we re-introduce mutable scope.

Comment (sgreene570) [reply pkg/operator/controller/ingress/load_balancer_service.go:329]: aha! That makes sense. Looking forward to taking advantage of that :wink:

**Pull request merged**